common/courses/notifications.ts

```diff
@@ -185,8 +185,47 @@ export const canPromoteSubcourse = async (subcourse: Prisma.subcourse, attempted
 export async function sendPupilCoursePromotion(subcourse: Prisma.subcourse, promotionType: Prisma.subcourse_promotion_type_enum) {
     const { allowed, reason } = await canPromoteSubcourse(subcourse, promotionType);
     if (!allowed) {
-        logger.info(`Can't promote Subcourse(${subcourse.id}). Reason: ${reason}`);
```

Comment:
I think we could leave this here and remove all the duplicates in each `if` statement :)